pynumaflow/mapper/_dtypes.py

```diff
@@ -162,6 +170,11 @@ def watermark(self) -> datetime:
         """Returns the watermark of the event."""
         return self._watermark
 
+    @property
+    def headers(self) -> dict[str, str]:
+        """Returns the headers of the event."""
+        return self._headers
```

Comment:
```suggestion
        return self._headers.copy()
```
To avoid unintentional mutation.

Reply:
Also let's cover this line in test?